datasets/cartesian/tensorflow-datasets/cartesian.py

```diff
@@ -0,0 +1,96 @@
+"""cartesian."""
+
+import sys
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+sys.path.append('../../datasets')
+from datasets.common_files.example_generator import example_generator
+
+
+_DESCRIPTION = """
+Original and iterative decoding data for the cartesian product dataset.
+"""
+
+_CITATION = """
+"""
+
+
+class Cartesian(tfds.core.GeneratorBasedBuilder):
+  """DatasetBuilder for cartesian dataset."""
+
+  MANUAL_DOWNLOAD_INSTRUCTIONS = """
+  Run datasets/cartesian/data_generation.py with the desired values for the flags 
+  predict_row, copy_output and short_input, and save the output files in 
+  `manual_dir/data`.
+  """
+
+  VERSION = tfds.core.Version('1.0.0')
+  RELEASE_NOTES = {
+      '1.0.0': 'Initial release.',
+  }
+
+  def _info(self) -> tfds.core.DatasetInfo:
+    """Returns the dataset metadata."""
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+          'source': tfds.features.Text(),
+          'target': tfds.features.Text(),
+          'op': tf.int32,
+        }),
+        supervised_keys=('source', 'target'), 
+        disable_shuffling=True,
+        homepage='',
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager: tfds.download.DownloadManager):
+    """Returns SplitGenerators."""
+    archive_path = dl_manager.manual_dir / 'data'
+    extracted_path = dl_manager.extract(archive_path)
+    return {
+      # original data
```

Comment:
nit: Use proper capitalization and punctuation for comments, e.g., here: "Original data.". Same for the others below